Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -491,6 +501,199 @@ std::unique_ptr<Document> Serializer::DecodeDocument(
       DocumentState::kSynced);
 }
 
+google_firestore_v1_Write Serializer::EncodeMutation(
+    const model::Mutation& mutation) const {
+  google_firestore_v1_Write result{};
+
+  if (!mutation.precondition().IsNone()) {
+    result.current_document = EncodePrecondition(mutation.precondition());
+  }
+
+  switch (mutation.type()) {
+    case Mutation::Type::kSet:
+      result.which_operation = google_firestore_v1_Write_update_tag;
+      result.update = EncodeDocument(
+          mutation.key(),
+          static_cast<const SetMutation&>(mutation).value().object_value());
+      return result;
+
+    case Mutation::Type::kPatch: {
+      result.which_operation = google_firestore_v1_Write_update_tag;
+      auto patch_mutation = static_cast<const PatchMutation&>(mutation);
+      result.update =
+          EncodeDocument(mutation.key(), patch_mutation.value().object_value());
+      result.update_mask = EncodeDocumentMask(patch_mutation.mask());
+      return result;
+    }
+
+      // TODO(rsgowman): Implement transform mutations. Probably like this:
+      /*
+      case Mutation::Type::kTransform:
+        result.which_operation = google_firestore_v1_Write_transform_tag;
+        auto transform = static_cast<const TransformMutation&>(mutation);
+        result.transform.document = EncodeKey(transform.key());
+
+        size_t count = transform.field_transforms.size();
+        result.transform.field_transforms_count = count;
+        result.transform.field_transforms =
+      MakeArray<google_firestore_v1_DocumentTransform_FieldTransform>(count);
+        int i = 0;
+        for (const FieldTransform& field_transform :
+      transform.field_transforms()) { result.transform.field_transforms[i] =
+      EncodeFieldTransform(field_transform); i++;
+        }
+        return result;
+      */
+
+    case Mutation::Type::kDelete:
```

Comment:
The style is not consistent. I know the kPatch has new local variable and asks for {. How about make every case use {.